lib/spack/spack/test/compilers/basics.py

```diff
@@ -863,7 +863,7 @@ class MockPackage(object):
 
     apple_clang_cls = spack.compilers.class_for_compiler_name("apple-clang")
     compiler = apple_clang_cls(
-        spack.spec.CompilerSpec("apple-clang@11.0.0"),
+        spack.spec.CompilerSpec.maybe_literal("apple-clang@11.0.0"),
```

Comment:
This was way more readable before we had to write `maybe_literal` -- why can't the constructor take either a `CompilerSpec` or a `str`?